src/Microsoft.DotNet.ProjectModel/OutputPathCalculator.cs

```diff
@@ -61,5 +74,58 @@ public string GetIntermediateOutputPath(string buildConfiguration, string interm
 
             return intermediateOutputPath;
         }
+        
+        public string GetAssemblyPath(string buildConfiguration)
+        {
+            var compilationOptions = _project.GetCompilerOptions(_framework, buildConfiguration);
+            var outputExtension = FileNameSuffixes.DotNet.DynamicLib;
+
+            if (_framework.IsDesktop() && compilationOptions.EmitEntryPoint.GetValueOrDefault())
+            {
+                outputExtension = FileNameSuffixes.DotNet.Exe;
+            }
+
+            return Path.Combine(
+                GetOutputDirectoryPath(buildConfiguration),
+                _project.Name + outputExtension);
+        }
+        
+        public IEnumerable<string> GetBuildOutputs(string buildConfiguration)
+        {
+            var assemblyPath = GetAssemblyPath(buildConfiguration);
+            
+            yield return assemblyPath;
+            yield return Path.ChangeExtension(assemblyPath, "pdb");
+
+            var compilationOptions = _project.GetCompilerOptions(_framework, buildConfiguration);
+            
+            if (compilationOptions.GenerateXmlDocumentation == true)
+            {
+                yield return Path.ChangeExtension(assemblyPath, "xml");
+            }
```

Comment:
It does matter a bit. If the compiler produces an exe and you delete it, it won't trigger a rebuild.
Or are you saying that on full framework the GetAssemblyPath and GetExecutablePath both return the same thing?